riscv-zig-blink/src/main.zig

```diff
@@ -0,0 +1,121 @@
+const builtin = @import("builtin");
+const std = @import("std");
+const fomu = @import("./fomu.zig");
+
+// export our _start symbol
+comptime {
+    @export(fomu.start._start, .{ .name = "_start" });
+}
+
+pub fn panic(message: []const u8, stack_trace: ?*builtin.StackTrace) noreturn {
+    @setCold(true);
+    while (true) {}
+}
+
+pub fn main() noreturn {
+    // Turn on the RGB block and current enable, as well as enabling led control
+    fomu.RGB.CTRL.* = .{
+        .EXE = true,
+        .CURREN = true,
+        .RGBLEDEN = true,
+        .RRAW = false,
+        .GRAW = false,
+        .BRAW = false,
+    };
+
+    // Enable the LED driver, and set 250 Hz mode.
+    // Also set quick stop, which we'll use to switch patterns quickly.
+    fomu.RGB.setControlRegister(.{
+        .enable = true,
+        .fr = .@"250",
+        .quick_stop = true,
+        .outskew = false,
+        .output_polarity = .active_high,
+        .pwm_mode = .linear,
+        .BRMSBEXT = 0,
+    });
+
+    // Set clock register to 12 MHz / 64 kHz - 1
+    fomu.RGB.setRegister(.BR, (12000000 / 64000) - 1);
+
+    // Blink on/off time is in 32 ms increments
+    fomu.RGB.setRegister(.ONR, 1); // Amount of time to stay "on"
+    fomu.RGB.setRegister(.OFR, 0); // Amount of time to stay "off"
+
+    fomu.RGB.setBreatheRegister(.On, .{
+        .enable = true,
+        .pwm_range_extend = false,
+        .mode = .fixed,
+        .rate = 1,
+    });
+    fomu.RGB.setBreatheRegister(.Off, .{
+        .enable = true,
+        .pwm_range_extend = false,
+        .mode = .fixed,
+        .rate = 1,
+    });
+
+    var i: u8 = 0;
+    while (true) : (i +%= 1) {
+        colorWheel(i);
+        msleep(80);
+    }
+}
+
+const Color = struct {
+    r: u8,
+    g: u8,
+    b: u8,
+};
+
+/// Input a value 0 to 255 to get a color value.
+/// The colours are a transition r - g - b - back to r.
+fn colorWheel(wheelPos: u8) void {
+    var c: Color = undefined;
+    var wp = 255 - wheelPos;
+    switch (wp) {
+        0...84 => {
+            c = .{
+                // .r = 255 - wp * 3,
```

Comment:
commented out for now as zig's compiler_rt is missing `__muldsi3` for riscv32
https://github.com/ziglang/zig/issues/3275 might make this harder...